app/src/main/java/com/github/bgabriel998/softwaredevproject/CameraPreview.java

```diff
@@ -0,0 +1,142 @@
+            try {
+                cameraProvider = cameraProviderFuture.get();
+                bindPreview(cameraProvider);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }, ContextCompat.getMainExecutor(context));
+    }
+
+    /**
+     * Declare and bind preview and analysis use cases
+     * @param cameraProvider used to bind the lifecycle of the camera
+     */
+    private void bindPreview(ProcessCameraProvider cameraProvider) {
+        //Get screen metrics
+        DisplayMetrics displayMetrics = new DisplayMetrics();
+        previewView.getDisplay().getRealMetrics(displayMetrics);
+
+        //Calculate aspectRatio
+        int screenAspectRatio = aspectRatio(displayMetrics.widthPixels, displayMetrics.heightPixels);
+
+        //Get screen rotation
+        int rotation = previewView.getDisplay().getRotation();
+
+        //CameraSelector
+        CameraSelector cameraSelector = new CameraSelector.Builder()
+                //Only use back facing camera
+                .requireLensFacing(CameraSelector.LENS_FACING_BACK)
+                .build();
+
+        //preview
+        Preview preview = new Preview.Builder()
+                //Set aspect ratio but not resolution, resolution is optimized by CameraX
+                .setTargetAspectRatio(screenAspectRatio)
+                //Set initial rotation
+                .setTargetRotation(rotation)
+                .build();
+
+        //ImageAnalysis
+        //Only deliver latest image to the analyzer
+        ImageAnalysis imageAnalysis = new ImageAnalysis.Builder()
+                .setTargetAspectRatio(screenAspectRatio)
+                .setTargetRotation(rotation)
+                //Only deliver latest image to the analyzer
+                //.setBackpressureStrategy(ImageAnalysis.STRATEGY_KEEP_ONLY_LATEST)
+                .build();
+        //Assign analyzer to the instance
+        imageAnalysis.setAnalyzer(cameraExecutor, ImageProxy::close);
+
+        //Unbind use-cases before rebinding
+        cameraProvider.unbindAll();
+
+        //Bind use cases to camera
+        cameraProvider.bindToLifecycle((LifecycleOwner) context, cameraSelector, preview, imageAnalysis);
+
+        //Attach the viewfinder's surface provider to preview
+        preview.setSurfaceProvider(previewView.getSurfaceProvider());
+    }
+
+    /**
+     * Calculate the aspect ratio of the display in function of the width and height of the screen
+     * @param width width of the preview in Pixels
+     * @param height height of the preview in Pixels
+     * @return Aspect ratio of the phone
+     */
+    private int aspectRatio(int width, int height){
+        double previewRatio = (double)Math.max(width, height)/Math.min(width, height);
+        double RATIO_16_9_VALUE = 16.0 / 9.0;
+        double RATIO_4_3_VALUE = 4.0 / 3.0;
+        if(Math.abs(previewRatio - RATIO_4_3_VALUE) <= Math.abs(previewRatio - RATIO_16_9_VALUE)){
```

Comment:
```suggestion
        return (Math.abs(previewRatio - RATIO_4_3_VALUE) <= Math.abs(previewRatio - RATIO_16_9_VALUE)) ? RATIO_4_3 : RATIO_16_9;
```
This way we could write all the remaining code of this method in just one line.
Everything else fine!